assets/course-theme/blocks/register-template-blocks.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * WordPress dependencies
+ */
+import { registerBlockType, unregisterBlockType } from '@wordpress/blocks';
+import { select, subscribe } from '@wordpress/data';
+
+/**
+ * Makes sure the template blocks are only registered when in the site or widget editor, or editing the template from
+ * the lesson page.
+ *
+ * @param {Array} blocks
+ */
+export function registerTemplateBlocks( blocks ) {
+	let themeBlocksEnabled = false;
+
+	const toggleBlockRegistration = ( enable ) => {
+		if ( enable === themeBlocksEnabled ) {
+			return;
+		}
+		themeBlocksEnabled = enable;
+		const method = enable ? registerBlockType : unregisterBlockType;
+		blocks.forEach( ( block ) => {
+			const { name, ...settings } = block;
+			method( name, settings );
+		} );
+	};
+
+	toggleBlockRegistration( true );
+
+	// TODO Only subscribe when in the post editor.
+	subscribe( () => {
+		const postType = select( 'core/editor' ).getCurrentPostType();
+
+		if ( ! postType ) {
+			return;
+		}
+
+		const { isEditingTemplate } = select( 'core/edit-post' );
+
+		const isTemplate = 'lesson' === postType && isEditingTemplate();
```

Comment:
You probably have a good reason why we don't include `quiz` here too. But I could not figure out on my own. 🤷 😄

Reply:
Yeah, the quiz can't be edited on its own for now, just inside the lesson, and the quiz editor even redirects to the lesson editor instead.

Reply:
Ah, of course. Makes sense.